src/components/slider/draw.js

```diff
@@ -0,0 +1,507 @@
+
+    labelItems.enter().append('g')
+        .classed(constants.labelGroupClass, true);
+
+    labelItems.exit().remove();
+
+    labelItems.each(function(d) {
+        var item = d3.select(this);
+
+        item.call(drawLabel, d, sliderOpts);
+
+        Lib.setTranslate(item,
+            normalizedValueToPosition(sliderOpts, d.fraction),
+            constants.tickOffset + constants.tickLength + sliderOpts.labelHeight
+        );
+    });
+
+}
+
+function handleInput(gd, sliderGroup, sliderOpts, normalizedPosition, doTransition) {
+    var quantizedPosition = Math.round(normalizedPosition * (sliderOpts.steps.length - 1));
+
+
+    if(quantizedPosition !== sliderOpts.active) {
+        setActive(gd, sliderGroup, sliderOpts, quantizedPosition, true, doTransition);
+    }
+}
+
+function setActiveByLabel(gd, sliderGroup, sliderOpts, label, doCallback, doTransition) {
+    var index;
+    for(var i = 0; i < sliderOpts.steps.length; i++) {
+        var step = sliderOpts.steps[i];
+        if(step.label === label) {
+            index = i;
+            break;
+        }
+    }
+
+    if(index !== undefined) {
+        setActive(gd, sliderGroup, sliderOpts, index, doCallback, doTransition);
+    }
+}
+
+function setActive(gd, sliderGroup, sliderOpts, index, doCallback, doTransition) {
+    sliderOpts._input.active = sliderOpts.active = index;
+
+    sliderGroup.call(setGripPosition, sliderOpts, sliderOpts.active / (sliderOpts.steps.length - 1), doTransition);
+
+    var step = sliderOpts.steps[sliderOpts.active];
+
+    if(step && step.method && doCallback) {
+        var args = step.args;
+        Plotly[step.method](gd, args[0], args[1], args[2]).catch(function() {
+            // This is not a disaster. Some methods like `animate` reject if interrupted
+            // and *should* nicely log a warning.
+            Lib.warn('Warning: Plotly.' + step.method + ' was called and rejected.');
+        });
+    }
+}
+
+function attachGripEvents(item, gd, sliderGroup, sliderOpts) {
+    var node = sliderGroup.node();
+    var $gd = d3.select(gd);
+
+    item.on('mousedown', function() {
+        var grip = sliderGroup.select('.' + constants.gripRectClass);
+
+        d3.event.stopPropagation();
+        d3.event.preventDefault();
+        grip.call(Color.fill, constants.gripBgActiveColor);
+
+        var normalizedPosition = positionToNormalizedValue(sliderOpts, d3.mouse(node)[0]);
+        handleInput(gd, sliderGroup, sliderOpts, normalizedPosition, true);
+
+        $gd.on('mousemove', function() {
+            var normalizedPosition = positionToNormalizedValue(sliderOpts, d3.mouse(node)[0]);
+            handleInput(gd, sliderGroup, sliderOpts, normalizedPosition, false);
+        });
+
+        $gd.on('mouseup', function() {
+            grip.call(Color.fill, constants.gripBgColor);
+            $gd.on('mouseup', null);
+            $gd.on('mousemove', null);
+        });
+    });
+}
+
+function drawTicks(sliderGroup, sliderOpts) {
```

Comment:
Good enough for now i.e. no need to make the slider tick options on par with axis tick options (++ `axes.js` is most fun module to reuse).